wig/bigWig/io.go

```diff
@@ -0,0 +1,141 @@
+	decodeBinaryField(file, &header.ExtensionOffset)
+	return header
+}
+
+func readZoomHeader(file *fileio.EasyReader) ZoomHeader {
+	var answer = ZoomHeader{}
+	decodeBinaryField(file, &answer.ReductionLevel)
+	decodeBinaryField(file, &answer.Reserved)
+	decodeBinaryField(file, &answer.DataOffset)
+	decodeBinaryField(file, &answer.IndexOffset)
+	return answer
+}
+
+func readTotalSummary(file *fileio.EasyReader) TotalSummaryBlock {
+	var answer = TotalSummaryBlock{}
+	decodeBinaryField(file, &answer.BasesCovered)
+	decodeBinaryField(file, &answer.MinVal)
+	decodeBinaryField(file, &answer.MaxVal)
+	decodeBinaryField(file, &answer.SumData)
+	decodeBinaryField(file, &answer.SumSquares)
+	return answer
+}
+
+func readChromTreeHeader(file *fileio.EasyReader) ChromTreeHeader {
+	var answer = ChromTreeHeader{}
+	decodeBinaryField(file, &answer.Magic)
+	if answer.Magic == chromTreeMagicBigEndian {
+		log.Fatalf("Error: bigWig package found a big endian chromosome tree header. Currently our package only supports little endian bigWig files.\n")
+	}
+	if answer.Magic != chromTreeMagic {
+		log.Fatalf("Error: Expected to find chromosome tree magic number (2026540177) in file. Found: %v.\n", answer.Magic)
+	}
+	decodeBinaryField(file, &answer.BlockSize)
+	decodeBinaryField(file, &answer.KeySize)
+	decodeBinaryField(file, &answer.ValSize)
+	decodeBinaryField(file, &answer.ItemCount)
+	decodeBinaryField(file, &answer.Reserved)
+	if answer.Reserved != 0 {
+		log.Fatalf("Error: expected chromosome tree header reserved field to be equal to 0. Found: %v.\n", answer.Reserved)
+	}
+	return answer
+}
+
+func readChromTreeNodes(file *fileio.EasyReader, header ChromTreeHeader) []ChromTreeNode {
+	var answer = make([]ChromTreeNode, 0)
+	var currNode ChromTreeNode
+	var currItem ChromTreeItem
+	var currByteCounter uint32
+	var countIdx uint16
+	numItems := 0
+
+	for numItems < int(header.ItemCount) {
+		// first we make a Node and read the node header
+		currNode = ChromTreeNode{}
+		decodeBinaryField(file, &currNode.IsLeaf)
+		decodeBinaryField(file, &currNode.Reserved)
+		decodeBinaryField(file, &currNode.Count)
+
+		for countIdx = 0; countIdx < currNode.Count; countIdx++ {
+			currItem = ChromTreeItem{}
+			// first we parse the key (same for leaf and non-leaf nodes, and comes first)
+			currItem.Key = make([]byte, header.KeySize)
+			for currByteCounter = 0; currByteCounter < header.KeySize; currByteCounter++ {
+				decodeBinaryField(file, &currItem.Key[currByteCounter])
+			}
+			if currNode.IsLeaf {
+				decodeBinaryField(file, &currItem.ChromId)
+				decodeBinaryField(file, &currItem.ChromSize)
+			} else {
+				decodeBinaryField(file, &currItem.ChildOffset)
+			}
+			currNode.Items = append(currNode.Items, currItem)
+			numItems++
+		}
+		answer = append(answer, currNode)
+	}
+
+	return answer
+}
+
+// decodeBinaryField reads binary data from an input *fileio.EasyReader to a variable.
+func decodeBinaryField(file *fileio.EasyReader, data any) {
```

Comment:
Looks good, I think it makes sense to put this in the fileio. I'd say it's potentially reusable by other future projects. We live in world of large data sets and binary operations on data is common practice.

Make sure this is covered by a small test. I'm happy to approve once that's complete. Something along the lines of:
```golang
func TestDecodeBinaryField(t *testing.T) {
	// Define test data
	var testData uint32 = 123

	binaryReader := fileio.EasyOpen(filename)

	// Call the function being tested
	var decodedData uint32
	decodeBinaryField(binaryReader, &decodedData)

	// Check if the decoded data matches the test data
	if decodedData != testData {
		t.Errorf("Error: Decoded data does not match test data. Expected: %d, Got: %d", testData, decodedData)
	}
}
```

Reply:
If you do opt to put this in fileio I would suggesting putting `LittleEndian` or `Le` or something into the name to denote the endianness.